backend/remote-state/azure/client.go

```diff
@@ -67,6 +70,17 @@ func (c *RemoteClient) Put(data []byte) error {
 	}
 
 	ctx := context.TODO()
+
+	if c.versioning {
+		snapshotInput := blobs.SnapshotInput{LeaseID: options.LeaseID}
+
+		if _, err := c.giovanniBlobClient.Snapshot(ctx, c.accountName, c.containerName, c.keyName, snapshotInput); err != nil {
```

Comment:
can we add a debug log before this too:

```suggestion
		log.Printf("[DEBUG] Snapshotting existing Blob %q (Container %q / Account %q)", c.keyName, c.containerName, c.accountName)
		if _, err := c.giovanniBlobClient.Snapshot(ctx, c.accountName, c.containerName, c.keyName, snapshotInput); err != nil {
```